sardana_albaem/ctrl/Albaem2OneDCtrl.py

```diff
@@ -0,0 +1,486 @@
+                return func(self, *args, **kwargs)
+            except Exception as e:
+                six.raise_from(RuntimeError(msg), e)
+        return wrapper
+
+
+class Albaem2OneDCtrl(OneDController):
+    MaxDevice = 5
+
+    ctrl_properties = {
+        'AlbaEmHost': {
+            Description: 'AlbaEm Host name',
+            Type: str
+        },
+        'Port': {
+            Description: 'AlbaEm Host name',
+            Type: int
+        },
+        'ExtTriggerInput': {
+            Description: 'ExtTriggerInput',
+            Type: str,
+            DefaultValue: "TRIGGER_IN",
+        },
+    }
+
+    ctrl_attributes = {
+        'AcquisitionMode': {
+            Type: str,
+            Description: 'Acquisition Mode: CHARGE, INTEGRATION',
+            Access: DataAccess.ReadWrite,
+            Memorize: Memorized,
+            FGet: "get_AcquisitionMode",
+            FSet: "set_AcquisitionMode"
+        },
+        'PointsPerStep': {
+            Type: int,
+            Description: "Points to generate or Triggers to expect per step. \
+                          Only applicable for the step scan. \
+                          For multiple points per step.",
+            Access: DataAccess.ReadWrite,
+            Memorize: Memorized,
+            FGet: "get_PointsPerStep",
+            FSet: "set_PointsPerStep"
+        },
+    }
+
+    axis_attributes = {
+        "Range": {
+            Type: str,
+            Description: 'Range for the channel',
+            Memorize: NotMemorized,
+            Access: DataAccess.ReadWrite,
+            FGet: "get_Range",
+            FSet: "set_Range",
+        },
+        "Inversion": {
+            Type: bool,
+            Description: 'Channel Digital inversion',
+            Memorize: NotMemorized,
+            Access: DataAccess.ReadWrite,
+            FGet: "get_Inversion",
+            FSet: "set_Inversion",
+
+        },
+        "InstantCurrent": {
+            Type: float,
+            Description: 'Channel instant current',
+            Memorize: NotMemorized,
+            Access: DataAccess.ReadOnly,
+            FGet: "get_InstantCurrent",
+        },
+        "FORMULA": {
+            Type: str,
+            Description: 'The formula to get the real value.\n '
+                            'e.g. "(value/10)*1e-06"',
+            Access: DataAccess.ReadWrite
+        },
+    }
+
+    @handle_error(msg="__init__: Could not connect to the device!")
+    def __init__(self, inst, props, *args, **kwargs):
+        """Class initialization."""
+        OneDController.__init__(self, inst, props, *args, **kwargs)
+        self.ip_config = (self.AlbaEmHost, self.Port)
+        self.albaem_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.albaem_socket.settimeout(1)
+        self.albaem_socket.connect(self.ip_config)
+        self.itime = 0.0
+        self.master = None
+        self._latency_time = 0.001  # In fact, it is just 320us
+        self._repetitions = 0
```

Comment:
what is repetitions for? and its relationship with pointsPerStep?

Reply:
should not be called NumberOfTriggers?

Reply:
this is internal sardana variable coming from base class!